app.py

```diff
@@ -49,3 +66,93 @@ def serve_angular(path):
         ])
         return flask.redirect(target)
     return flask.send_file('dist/client/index.html')
+
+@app.route('/login')
+def login():
+    """'Login' page for auth flow"""
+    return '<a href="/auth">Authorize</a>'
+
+@app.route('/auth')
+def oauth_flow():
+    """Redirect for user authorization"""
+    genius = OAuth2Service(
+        client_id=CLIENT_ID,
+        client_secret=CLIENT_SECRET,
+        name='genius',
+        authorize_url=AUTHORIZE_URL,
+        access_token_url=ACCESS_TOKEN_URL,
+        base_url=BASE_URL)
+
+    redirect_uri = REDIRECT_URI
+
+    # TODO Define a better state
+    state = 'state'
+
+    params = {
+        'scope': 'me',
+        'response_type': 'code',
+        'redirect_uri': redirect_uri,
+        'state': state}
+
+    authorize_url = genius.get_authorize_url(**params)
+
+    return flask.redirect(authorize_url)
+
+@app.route('/in')
+def auth_redirect():
+    """_Authorization successful_; display song lyrics to test acces token"""
+    genius = OAuth2Service(
+        client_id=CLIENT_ID,
+        client_secret=CLIENT_SECRET,
+        name='genius',
+        authorize_url=AUTHORIZE_URL,
+        access_token_url=ACCESS_TOKEN_URL,
+        base_url=BASE_URL)
+
+    session = genius.get_auth_session(
+        data={
+            'code': flask.request.args.get('code'),
+            'redirect_uri': REDIRECT_URI,
+            'grant_type': 'authorization_code'},
+        decoder=oauth_decode)
+
+    response = session.get(
+        'songs/{id}'.format(id=1929408))
+
+    lyrics = response.json()['response']['song']['embed_content']
+
+    audio_snippet = response.json()['response']['song'] \
+    ['apple_music_player_url']
```

Comment:
possibly more readable (also possibly more efficient, given that response.json() is a method and might be doing work or changing things to get its result)

```suggestion
    song = response.json()
    lyrics = song['embed_content']

    audio_snippet = song['apple_music_player_url']
```